crypto/bio/bss_mem.c

```diff
@@ -192,11 +197,15 @@ static int mem_read(BIO *b, char *out, int outl)
     BIO_BUF_MEM *bbm = (BIO_BUF_MEM *)b->ptr;
     BUF_MEM *bm = bbm->readp;
 
+    if (b->flags & BIO_FLAGS_MEM_RDONLY) {
+        bm = bbm->buf;
+    }
     BIO_clear_retry_flags(b);
     ret = (outl >= 0 && (size_t)outl > bm->length) ? (int)bm->length : outl;
     if ((out != NULL) && (ret > 0)) {
         memcpy(out, bm->data, ret);
         bm->length -= ret;
+        bm->max -= ret;
```

Comment:
is this an attempt to fix BIO_FLAGS_NONCLEAR_RST ?

Reply:
No, it's just kind-of attempt to keep the bm->max somehow correct in regards to the data pointer in case you obtain the BUF_MEM from the RDONLY BIO via the BIO_get_mem_ptr. Not that it really matters much as the read only BUF_MEM cannot be written to anyway.